dotcom-rendering/src/layouts/lib/decideAdSlots.tsx

```diff
@@ -0,0 +1,152 @@
+				<AdSlot
+					data-print-layout="hide"
+					position="merchandising-high"
+					hasPageskin={hasPageSkin}
+				/>
+			</Hide>
+		) : (
+			<AdSlot
+				data-print-layout="hide"
+				position="merchandising-high"
+				hasPageskin={hasPageSkin}
+			/>
+		);
+	} else if (mobileAdPositions.includes(index)) {
+		return (
+			<Hide from="tablet">
+				<AdSlot
+					index={mobileAdPositions.indexOf(index)}
+					data-print-layout="hide"
+					position="mobile-front"
+				/>
+			</Hide>
+		);
+	}
+
+	return null;
+};
+
+/**
+ * Renders a fronts-banner ad when in the fronts banner AB test.
+ * Only applies to network fronts on desktop screens and wider.
+ */
+export const decideFrontsBannerAdSlot = (
+	renderAds: boolean,
+	hasPageSkin: boolean,
+	numBannerAdsInserted: React.MutableRefObject<number>,
+	showBannerAds: boolean,
+	index: number,
+	pageId: string,
+	collectionName: string | null,
+	isInFrontsBannerTest?: boolean,
+	targetedCollections?: string[],
+) => {
+	const isFirstContainer = index === 0;
+	if (!renderAds || hasPageSkin || isFirstContainer) {
+		return null;
+	}
+
+	// The fronts banner 0% AB test has concluded. However, it still exists so that
+	// the commercial team can opt in and test ad campaigns against the live site.
+	// In this test, fronts-banner ads are inserted above specific collections and pages.
+	//
+	// This code block can be deleted when fronts banner ads are live and the FrontsBannerTest is removed.
+	if (isInFrontsBannerTest && collectionName) {
+		if (targetedCollections?.includes(collectionName)) {
+			numBannerAdsInserted.current = numBannerAdsInserted.current + 1;
+
+			return (
+				<AdSlot
+					data-print-layout="hide"
+					position="fronts-banner"
+					index={numBannerAdsInserted.current}
+					hasPageskin={hasPageSkin}
+				/>
+			);
+		}
+
+		// If the showBannerAds feature switch is on and the page was included in the AB test, then
+		// show ads above the collections as specified in that AB test. One reason is that on /uk
+		// the "Ukraine invasion" collection is third and we don't want to place an ad above this
+		// container, which means we don't get a banner ad until the 6th collection.
+	} else if (
+		showBannerAds &&
+		collectionName &&
+		Object.keys(frontsBannerAdCollections).includes(pageId)
+	) {
+		if (targetedCollections?.includes(collectionName)) {
+			numBannerAdsInserted.current = numBannerAdsInserted.current + 1;
+
+			return (
+				<AdSlot
+					data-print-layout="hide"
+					position="fronts-banner"
+					index={numBannerAdsInserted.current}
+					hasPageskin={hasPageSkin}
+				/>
+			);
+		}
+	}
+
+	// Insert an ad after every third collection. Warning: may skip an ad if a collection isn't rendered.
+	// e.g. if the 15th collection doesn't render, an ad is shown above the 12th and the 18th, skipping the 15th.
+	else if (
+		showBannerAds &&
+		numBannerAdsInserted.current < MAX_FRONTS_BANNER_ADS &&
+		index % 3 === 2 && // Insert above the 3rd, 6th, ..., container
```

Comment:
This
```suggestion
		(index  + 1) % 3 === 0 && // Insert above the 3rd, 6th, ..., container
```
would be clearer in my head, but not a blocker